Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -430,12 +434,22 @@ private static function _getResultMessageArray($sType, $sMessage)
     private function _getTransactionHandler()
     {
         // NOTE: if _oTransactionHandler got injected for testing, use it
-        if (is_null($this->_oTransactionHandler)) {
-            $oConfig = $this->_getPaymentMethodConfig();
-            $this->_oTransactionHandler = new TransactionHandler($this->_getBackendService($oConfig));
+        if ($this->_oTransactionHandler) {
+            return $this->_oTransactionHandler;
         }
 
-        return $this->_oTransactionHandler;
+        $oConfig = $this->_getPaymentMethodConfig();
+
+        $oSepaCtPayment = PaymentMethodHelper::getPaymentById(SepaCreditTransferPaymentMethod::getName(true));
```

Comment:
This is in my opinion bad practice. As now the SEPA CT config is added all the time. Also this code is not belonging here, here the only thing that should be is $oConfig = $this->_getPaymentMethodConfig(); all other stuff needs to happen in payment methods. Please solve this issue with inheritance or a different class structure. Maybe we can solve it with adding a interface or with another technique.  Then we can have sperate implementations of one where only payment method Config is loaded and one where we load a SEPA CT config if its a refund. If you need help please contact me, but i think with this we just shifting the responsibility to a wrong place.

Reply:
I changed it, so all payment methods which need sepa ct for refund are inherited from sepa ct